### PR TITLE
Institution Program Search

### PR DESCRIPTION
**Pull request by devin-mccurdy**
## Description
VET TEC search now searches and displays GIDS InstitutionPrograms results instead of Institutions (which are still used for non-VET TEC searches). All changes are made behind a prod flag.

As a Comparison Tool user, I need the ability to see the VET TEC search results base on Program so that I can see information related to the course of study that I am interested in pursuing.

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/19734

## Testing done
Unit test coverage, QA review

## Screenshots
image

## Acceptance criteria
- [x] When the "Type of Institution" is "VET TEC training providers" and the user clicks "Search Schools", the search results displays all available VET TEC programs.
- [x] The provider name is still displayed as part of the search results.
- [x] The provider city, state is still displayed as part of the search results.
- [x] The "Tuition" amount for the program is displayed.
- [x] The "Length" of the program in weeks is displayed.
- [x] The "Housing" of the program is displayed based on the lowest rate between VA/DOD.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi/components/vet-tec/VetTecProgramSearchResult.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+              <div className="small-12 usa-width-seven-twelfths medium-7 columns">
+                <h2>
+                  <Link to={linkTo}>{description}</Link>
+                </h2>
+                <div style={{ position: 'relative', bottom: 0 }}>
+                  <p className="institution-name vads-u-font-weight--bold">
+                    {institutionName}
+                  </p>
+                  <p className="institution-location">{`${city}, ${state ||
+                    country}`}</p>
```
Comment (shawkey-daniel) [inline src/applications/gi/components/vet-tec/VetTecProgramSearchResult.jsx]: SearchResult.jsx is using locationInfo from helpers.js to handle possible empty city/state/country values. Not pretty, but avoids leading or trailing commas.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]

File: src/applications/gi/actions/index.js [hunk 2]
```diff
@@ -167,7 +168,32 @@ export function fetchSearchResults(query = {}) {
...
+  };
+}
+
+export function fetchProgramSearchResults(query = {}) {
+  const queryString = Object.keys(query).reduce((str, key) => {
+    const sep = str ? '&' : '';
+    return `${str}${sep}${snakeCase(key)}=${query[key]}`;
+  }, '');
+
+  const url = `${api.url}/institution_programs/search?${queryString}`;
```
Comment (U-DON) [inline src/applications/gi/actions/index.js]: Would importing `appendQuery` from the `append-query` module serve your needs here?

File: src/applications/gi/components/vet-tec/VetTecProgramSearchResult.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      pathname: `profile/${facilityCode}/${description}`,
+      query: version ? { version } : {},
+    };
+
+    return (
+      <div className="search-result">
+        <div className="outer">
+          <div className="inner">
+            <div className="row">
+              <div className="small-12 usa-width-seven-twelfths medium-7 columns">
```
Comment (U-DON) [inline src/applications/gi/components/vet-tec/VetTecProgramSearchResult.jsx]: I would suggest using the [layout classes from the design system](https://design.va.gov/layout/flexbox-grid) whenever possible, but I understand if the surrounding markup is still using the Foundation classes.

File: src/applications/gi/reducers/utility.js [hunk 4]
```diff
@@ -25,3 +25,25 @@ export const normalizedAttributes = attributes => {
...
+    : attributes.description;
+  const city = attributes.city
+    ? attributes.city.toUpperCase()
+    : attributes.city;
+  const state = attributes.state
+    ? attributes.state.toUpperCase()
+    : attributes.state;
+  const country = attributes.country
+    ? attributes.country.toUpperCase()
+    : attributes.country;
```
Comment (U-DON) [inline src/applications/gi/reducers/utility.js]: These could be made a bit more succinct if you'd like:

```
const description = attributes.description && attributes.description.toUpperCase();
```

**Review by U-DON — APPROVED** [review]

File: src/applications/gi/actions/index.js (hunk 2 above)
Comment (dmccurdybah) [reply src/applications/gi/actions/index.js]: Yes, thanks!

**Pull request merged**